app/views/Register/RegisterForm/common.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    ObjectSchema,
+    PartialForm,
+    emailCondition,
+    requiredCondition,
+    requiredStringCondition,
+    lengthGreaterThanCondition,
+    lengthSmallerThanCondition,
+} from '@togglecorp/toggle-form';
+
+import { UserUserType } from '#generated/types';
+
+interface BaseExtraFields {
+    name: string;
+    municipality: string;
+    wardNumber: number;
+    localAddress: string;
+    panNumber: string;
+    vatNumber: string;
+}
+
+interface InstitutionFields extends BaseExtraFields {
+}
+
+interface PublisherFields extends BaseExtraFields {
+}
+
+interface SchoolFields extends BaseExtraFields {
+}
+
+export interface RegistrationFields {
+    email: string;
+    firstName?: string;
+    lastName?: string;
+    password: string;
+    userType: UserUserType;
+    phoneNumber: string;
+    institution?: InstitutionFields;
+    publisher?: PublisherFields;
+    school?: SchoolFields;
+}
```

Comment:
Let's use interfaces from server

Reply:
It's slightly different from the current usecase